Quality_control.do

```diff
@@ -14,31 +16,33 @@
     }
 	
 	***for variables generated from 4_sexual_health
-	foreach var of var w_CPR w_unmet_fp	w_need_fp w_metany_fp w_metmod_fp w_metany_fp_q {
+	foreach var of var w_CPR w_unmet_fp	w_need_fp w_metany_fp w_metmod_fp w_metany_fp_q w_condom_conc {
 	egen pop_`var' = wtmean(`var'), weight(w_sampleweight)
 	}
 	
 	***for variables generated from 7_child_vaccination
 	foreach var of var c_bcg c_dpt1 c_dpt2 c_dpt3 c_fullimm c_measles ///
 	c_polio1 c_polio2 c_polio3{
-    egen pop_`var' = wtmean(`var'), weight(w_sampleweight)
+    egen pop_`var' = wtmean(`var'), weight(hh_sampleweight)
```

Comment:
sure.
the problem here is that the weight variables, both women and children, are missing for some of the indicator variables, such as mor_abe, and therefore when generating the indicator file will return missing value. i was thinking using hhweight instead, which is in most of the time, not missing. 
however, in some of the cases the hh_weight, household weight, might not be appropriate for generating individual variables.

Reply:
@robin-wang

Reply:
Hi @Ortsang , 

Thanks in ton for your response.

I went back to the raw data. And found that the missing weight issue might be the problem of the template on the generation of w_sampleweight. 

In the Master do file, both child sample weight and women sample weight used w_sampleweight and merged together in the final output. However, the child sample weight (chweight) and women sample weight (wmweight) should be separately generated instead. Please let me know if using these two sample weights separately in response to the indicators could solve the issue you talked about (missing sample weight for some indicators), if so we should adjust the template. 

Regards, 
Aline

Reply:
Catching up on this thread, I was separately reaching out to Ke about working out why our quality indicator output suffers from many missing values. This is despite having SD in place.

Reply:
> Hi @Ortsang ,
> 
> Thanks in ton for your response.
> 
> I went back to the raw data. And found that the missing weight issue might be the problem of the template on the generation of w_sampleweight.
> 
> In the Master do file, both child sample weight and women sample weight used w_sampleweight and merged together in the final output. However, the child sample weight (chweight) and women sample weight (wmweight) should be separately generated instead. Please let me know if using these two sample weights separately in response to the indicators could solve the issue you talked about (missing sample weight for some indicators), if so we should adjust the template.
> 
> Regards, Aline

Hi Aline,

sure thing.
thank you for pointing that out, i went back to the master code and raw data and got the sense of how it works. as you said, the children and women sample are merged together and therefore, the number of sample weight is much less than the total number of observations. In the case of tonga2019_Adept, there is around 1100 sampleweight but in total there are more than 14000 observations.

i agree that we should use the w_weight instead since it is the weight of individual but not household.

referring to the question @robin-wang raised.
there is minor mismatch happening in the mortality code. in the data of creating mortality information, *bh, the only individual weight is wmweight. and there is no code generating weight for this subsample from *bh file. I think this is the reason that some of the mortality variable has no number while the variable sd has number.
maybe we could add a code adding weight within mortality?

best,
ke

Reply:
eureka!
the main problem here is that the weight varibale - w_sampleweight was not kept at the end of mortality.

code has been updated.

Reply:
Voila! Case solved!
That makes sense, because we typically assume values and SDs go hand-in-hand for most surveys we processed so far. That's a great update, and will roll out to the rest of this survey patch thanks to your help!

I'll replicate when you are ready to push then : )

Reply:
: ) cheers!